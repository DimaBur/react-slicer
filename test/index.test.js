import React from 'react'
import renderer from 'react-test-renderer'
import enzyme, {shallow} from 'enzyme'

import Slicer from '../src/index'

import Adapter from 'enzyme-adapter-react-16'

enzyme.configure({adapter: new Adapter()})

const defaultComponent = <Slicer
  itemsPerPage={3}
>
  <div key={1}>Item 1</div>
  <div key={2}>Item 2</div>
  <div key={3}>Item 3</div>
  <div key={4}>Item 4</div>
  <div key={5}>Item 5</div>
  <div key={6}>Item 6</div>
  <div key={7}>Item 7</div>
  <div key={8}>Item 8</div>
</Slicer>

test('Render without crashing', () => {
  const component = renderer.create(defaultComponent)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Pages changing', () => {
  const component = shallow(defaultComponent)
  expect(component.html()).toMatchSnapshot()
  component.instance().handleClickArrow('next')
  expect(component.html()).toMatchSnapshot()
  component.instance().handleClickArrow('next')
  expect(component.html()).toMatchSnapshot()
  component.instance().handleClickArrow('prev')
  expect(component.html()).toMatchSnapshot()
})

test('Items per page amount', () => {
  const component = shallow(defaultComponent)

  component.setState({
    itemsPerPage: 1
  })

  expect(component.instance().renderItems()).toHaveLength(1)

  component.setState({
    itemsPerPage: 4
  })

  expect(component.instance().renderItems()).toHaveLength(4)

  component.setState({
    itemsPerPage: 10
  })

  expect(component.instance().renderItems()).toHaveLength(8)
})
