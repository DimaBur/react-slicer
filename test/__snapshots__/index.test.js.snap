// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages changing 1`] = `"<div class=\\"react-slicer\\"><div class=\\"react-slicer__view\\"><div>Item 1</div><div>Item 2</div><div>Item 3</div></div><ul class=\\"react-slicer__pagination\\"><li class=\\"react-slicer__pagination-item react-slicer__pagination-item_prev\\"><span class=\\"react-slicer__arrow react-slicer__arrow_prev\\">&lt;</span></li><li class=\\"react-slicer__pagination-item react-slicer__pagination-item_active\\" id=\\"num-1\\">1</li><li class=\\"react-slicer__pagination-item\\" id=\\"num-2\\">2</li><li class=\\"react-slicer__pagination-item\\" id=\\"num-3\\">3</li><li class=\\"react-slicer__pagination-item react-slicer__pagination-item_next\\"><span class=\\"react-slicer__arrow react-slicer__arrow_next\\">&gt;</span></li></ul></div>"`;

exports[`Pages changing 2`] = `"<div class=\\"react-slicer\\"><div class=\\"react-slicer__view\\"><div>Item 1</div><div>Item 2</div><div>Item 3</div></div><ul class=\\"react-slicer__pagination\\"><li class=\\"react-slicer__pagination-item react-slicer__pagination-item_prev\\"><span class=\\"react-slicer__arrow react-slicer__arrow_prev\\">&lt;</span></li><li class=\\"react-slicer__pagination-item react-slicer__pagination-item_active\\" id=\\"num-1\\">1</li><li class=\\"react-slicer__pagination-item\\" id=\\"num-2\\">2</li><li class=\\"react-slicer__pagination-item\\" id=\\"num-3\\">3</li><li class=\\"react-slicer__pagination-item react-slicer__pagination-item_next\\"><span class=\\"react-slicer__arrow react-slicer__arrow_next\\">&gt;</span></li></ul></div>"`;

exports[`Pages changing 3`] = `"<div class=\\"react-slicer\\"><div class=\\"react-slicer__view\\"><div>Item 1</div><div>Item 2</div><div>Item 3</div></div><ul class=\\"react-slicer__pagination\\"><li class=\\"react-slicer__pagination-item react-slicer__pagination-item_prev\\"><span class=\\"react-slicer__arrow react-slicer__arrow_prev\\">&lt;</span></li><li class=\\"react-slicer__pagination-item react-slicer__pagination-item_active\\" id=\\"num-1\\">1</li><li class=\\"react-slicer__pagination-item\\" id=\\"num-2\\">2</li><li class=\\"react-slicer__pagination-item\\" id=\\"num-3\\">3</li><li class=\\"react-slicer__pagination-item react-slicer__pagination-item_next\\"><span class=\\"react-slicer__arrow react-slicer__arrow_next\\">&gt;</span></li></ul></div>"`;

exports[`Pages changing 4`] = `"<div class=\\"react-slicer\\"><div class=\\"react-slicer__view\\"><div>Item 1</div><div>Item 2</div><div>Item 3</div></div><ul class=\\"react-slicer__pagination\\"><li class=\\"react-slicer__pagination-item react-slicer__pagination-item_prev\\"><span class=\\"react-slicer__arrow react-slicer__arrow_prev\\">&lt;</span></li><li class=\\"react-slicer__pagination-item react-slicer__pagination-item_active\\" id=\\"num-1\\">1</li><li class=\\"react-slicer__pagination-item\\" id=\\"num-2\\">2</li><li class=\\"react-slicer__pagination-item\\" id=\\"num-3\\">3</li><li class=\\"react-slicer__pagination-item react-slicer__pagination-item_next\\"><span class=\\"react-slicer__arrow react-slicer__arrow_next\\">&gt;</span></li></ul></div>"`;

exports[`Render with custom class 1`] = `
<div
  className="custom-class"
>
  <div
    className="custom-class__view"
  >
    <div>
      Item 1
    </div>
    <div>
      Item 2
    </div>
    <div>
      Item 3
    </div>
  </div>
  <ul
    className="custom-class__pagination"
  >
    <li
      className="custom-class__pagination-item custom-class__pagination-item_prev"
      onClick={[Function]}
    >
      <span
        className="react-slicer__arrow react-slicer__arrow_prev"
      >
        &lt;
      </span>
    </li>
    <li
      className="custom-class__pagination-item custom-class__pagination-item_active"
      id="num-1"
      onClick={[Function]}
    >
      1
    </li>
    <li
      className="custom-class__pagination-item"
      id="num-2"
      onClick={[Function]}
    >
      2
    </li>
    <li
      className="custom-class__pagination-item"
      id="num-3"
      onClick={[Function]}
    >
      3
    </li>
    <li
      className="custom-class__pagination-item custom-class__pagination-item_next"
      onClick={[Function]}
    >
      <span
        className="react-slicer__arrow react-slicer__arrow_next"
      >
        &gt;
      </span>
    </li>
  </ul>
</div>
`;

exports[`Render without crashing 1`] = `
<div
  className="react-slicer"
>
  <div
    className="react-slicer__view"
  >
    <div>
      Item 1
    </div>
    <div>
      Item 2
    </div>
    <div>
      Item 3
    </div>
  </div>
  <ul
    className="react-slicer__pagination"
  >
    <li
      className="react-slicer__pagination-item react-slicer__pagination-item_prev"
      onClick={[Function]}
    >
      <span
        className="react-slicer__arrow react-slicer__arrow_prev"
      >
        &lt;
      </span>
    </li>
    <li
      className="react-slicer__pagination-item react-slicer__pagination-item_active"
      id="num-1"
      onClick={[Function]}
    >
      1
    </li>
    <li
      className="react-slicer__pagination-item"
      id="num-2"
      onClick={[Function]}
    >
      2
    </li>
    <li
      className="react-slicer__pagination-item"
      id="num-3"
      onClick={[Function]}
    >
      3
    </li>
    <li
      className="react-slicer__pagination-item react-slicer__pagination-item_next"
      onClick={[Function]}
    >
      <span
        className="react-slicer__arrow react-slicer__arrow_next"
      >
        &gt;
      </span>
    </li>
  </ul>
</div>
`;

exports[`Setting page 1`] = `"<div class=\\"react-slicer\\"><div class=\\"react-slicer__view\\"><div>Item 1</div><div>Item 2</div><div>Item 3</div></div><ul class=\\"react-slicer__pagination\\"><li class=\\"react-slicer__pagination-item react-slicer__pagination-item_prev\\"><span class=\\"react-slicer__arrow react-slicer__arrow_prev\\">&lt;</span></li><li class=\\"react-slicer__pagination-item react-slicer__pagination-item_active\\" id=\\"num-1\\">1</li><li class=\\"react-slicer__pagination-item\\" id=\\"num-2\\">2</li><li class=\\"react-slicer__pagination-item\\" id=\\"num-3\\">3</li><li class=\\"react-slicer__pagination-item react-slicer__pagination-item_next\\"><span class=\\"react-slicer__arrow react-slicer__arrow_next\\">&gt;</span></li></ul></div>"`;

exports[`Setting page 2`] = `"<div class=\\"react-slicer\\"><div class=\\"react-slicer__view\\"><div>Item 1</div><div>Item 2</div><div>Item 3</div></div><ul class=\\"react-slicer__pagination\\"><li class=\\"react-slicer__pagination-item react-slicer__pagination-item_prev\\"><span class=\\"react-slicer__arrow react-slicer__arrow_prev\\">&lt;</span></li><li class=\\"react-slicer__pagination-item react-slicer__pagination-item_active\\" id=\\"num-1\\">1</li><li class=\\"react-slicer__pagination-item\\" id=\\"num-2\\">2</li><li class=\\"react-slicer__pagination-item\\" id=\\"num-3\\">3</li><li class=\\"react-slicer__pagination-item react-slicer__pagination-item_next\\"><span class=\\"react-slicer__arrow react-slicer__arrow_next\\">&gt;</span></li></ul></div>"`;
